src/PackagePublisher/Program.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.CommandLineUtils;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace PackagePublisher
+{
+    public static class Program
+    {
+        private const string ApiKeyEnvironmentVariableName = "PACKAGEPUBLISHER_APIKEY";
+        private static CommandOption _packagesDirectory;
+        private static CommandOption _packagePath;
+        private static CommandOption _feedToUploadTo;
+        private static CommandOption _maxRetryCountOption;
+        private static IEnumerable<PackageInfo> _packages;
+
+        // Do NOT log the key
+        private static string _apiKey = null;
+        private static int _maxRetryCount = 10;
+
+        public static int Main(string[] args)
+        {
+            var app = new CommandLineApplication();
+            _packagesDirectory = app.Option(
+                "-d|--directory",
+                "The directory containing .nupkg file(s)",
+                CommandOptionType.SingleValue);
+            _packagePath = app.Option(
```

Comment:
I thought we landed on leaving Coherence along and just creating a tool that pushed packages in a dir?